src/core/execute.c

```diff
@@ -4909,6 +4910,16 @@ static int exec_child(
                         return r;
         }
 
+	if (context->memory_ksm) {
+		r = prctl(PR_SET_MEMORY_MERGE, 1);
+                if (r == -EOPNOTSUPP)
```

Comment:
nah, prctl() returns negative and the error code in errno, use this here:


```c
if (prctl(…) < 0) {
        if (ERRNO_IS_NOT_SUPPORTED(errno)) 
                log_unit_debug_errno(…);
        else {
                *exit_statues = EXIT_KSM;
                return log_unit_error_errno(…);
        }
}

Reply:
I made the above change.